plugins/UM3NetworkPrinting/src/Models/Http/ClusterPrinterConfigurationMaterial.py

```diff
@@ -24,32 +25,27 @@ def __init__(self, brand: Optional[str] = None, color: Optional[str] = None, gui
         self.material = material
         super().__init__(**kwargs)
 
-    ## Creates a material output model based on this cloud printer material.
+    ##  Creates a material output model based on this cloud printer material.
+    #
+    #   A material is chosen that matches the current GUID. If multiple such
+    #   materials are available, read-only materials are preferred and the
+    #   material with the earliest alphabetical name will be selected.
+    #   \return A material output model that matches the current GUID.
     def createOutputModel(self) -> MaterialOutputModel:
-        material_manager = CuraApplication.getInstance().getMaterialManager()
-        material_group_list = material_manager.getMaterialGroupListByGUID(self.guid) or []
-
-        # Sort the material groups by "is_read_only = True" first, and then the name alphabetically.
-        read_only_material_group_list = list(filter(lambda x: x.is_read_only, material_group_list))
-        non_read_only_material_group_list = list(filter(lambda x: not x.is_read_only, material_group_list))
-        material_group = None
-        if read_only_material_group_list:
-            read_only_material_group_list = sorted(read_only_material_group_list, key = lambda x: x.name)
-            material_group = read_only_material_group_list[0]
-        elif non_read_only_material_group_list:
-            non_read_only_material_group_list = sorted(non_read_only_material_group_list, key = lambda x: x.name)
-            material_group = non_read_only_material_group_list[0]
-
-        if material_group:
-            container = material_group.root_material_node.getContainer()
-            color = container.getMetaDataEntry("color_code")
-            brand = container.getMetaDataEntry("brand")
-            material_type = container.getMetaDataEntry("material")
-            name = container.getName()
+        container_registry = ContainerRegistry.getInstance()
+        same_guid = container_registry.findInstanceContainersMetadata(GUID = self.guid)
+        if same_guid:
+            read_only = sorted(filter(lambda metadata: container_registry.isReadOnly(metadata["id"]), same_guid), key = lambda metadata: metadata["name"])
+            if read_only:
+                material_metadata = read_only[0]
+            else:
+                material_metadata = min(same_guid, key = lambda metadata: metadata["name"])
         else:
-            color = self.color
-            brand = self.brand
-            material_type = self.material
-            name = "Empty" if self.material == "empty" else "Unknown"
-
-        return MaterialOutputModel(guid=self.guid, type=material_type, brand=brand, color=color, name=name)
+            material_metadata = {
+                "color_code": self.color,
+                "brand": self.brand,
+                "material": self.material,
+                "name": "Empty" if self.material == "empty" else "Unknown"
```

Comment:
This is correct. The "Empty" and "Unknown" are user-visible strings. I'm guessing they are not translated because they are listed in between the profile names and those are not translated, so this makes it look more consistent. I left the code here as-is but needed to move it into a dictionary since it uses the ContainerRegistry to get the metadata as dictionary too, and these lines are a fallback for that. It's not in Cura's place to refactor the code of the Connected Services team (but it is polite to repair their code if we break something for them).